includes/model/Openedx_Woocommerce_Plugin_Enrollment.php

```diff
@@ -207,51 +240,95 @@ public function insert_new( $enrollment_arr, $enrollment_action = '' ) {
      * @param array  $enrollment_arr An array containing the enrollment info.
      * @param string $enrollment_action The API action to perform once the wp process is done.
      */
-    public function save_enrollment( $post, $enrollment_arr, $enrollment_action ) {
+    public function save_enrollment($post, $enrollment_arr, $enrollment_action)
+    {
 
         $post_id = $post->ID;
 
-        $enrollment_course_id    = $enrollment_arr['enrollment_course_id'];
-        $enrollment_email        = $enrollment_arr['enrollment_email'];
-        $enrollment_mode         = $enrollment_arr['enrollment_mode'];
-        $enrollment_request_type = $enrollment_arr['enrollment_request_type'];
-        $enrollment_order_id     = $enrollment_arr['enrollment_order_id'];
+        // Prepare enrollment data function call
+        $data = $this->prepare_enrollment_data($post_id, $enrollment_arr);
 
-        // Get the old post metadata.
-        $old_course_id           = get_post_meta($post_id, 'course_id', true);
-        $old_email            = get_post_meta($post_id, 'email', true);
-        $old_mode                = get_post_meta($post_id, 'mode', true);
+        // Split returned arrays into 2 variables for old and current data.
+        $old_data = $data['old_data_array'];
+        $enrollment_data = $data['enrollment_arr'];
 
-        // We need to have all 3 required params to continue.
-        $enrollment_user_reference = $enrollment_email;
-        if ( ! $enrollment_course_id || ! $enrollment_user_reference || ! $enrollment_mode ) {
+        // Check if the enrollment main data is empty.
+        if (
+            !$enrollment_data['enrollment_course_id']
+            || !$enrollment_data['enrollment_email']
+            || !$enrollment_data['enrollment_mode']
+        ) {
             return;
         }
 
-        // Update the $post metadata.
-        update_post_meta( $post_id, 'course_id', $enrollment_course_id );
-        update_post_meta( $post_id, 'email', $enrollment_email );
-        update_post_meta( $post_id, 'mode', $enrollment_mode );
-        update_post_meta( $post_id, 'order_id', $enrollment_order_id );
+        // Update post meta data for the Enrollment.
+        $this->update_enrollment_meta_data($post_id, $enrollment_data);
+
+        /*
+         * Check if old post_meta tags are different from the new ones to
+         * change Enrollment Request name in Enrollment Manager requests list.
+         */
 
-        if ( $enrollment_request_type === 'enroll' ) {
-            update_post_meta( $post_id, 'is_active', true );
+        if (
```

Comment:
Same here can you extract this logic into separate method and call that method here